build_and_test.sh

```diff
@@ -0,0 +1,18 @@
+#! /bin/bash
+mkdir dist
```

Comment:
Would you try switching this file to just the following, since we discussed it and it works?
```
#! /bin/bash
hcdev test  | test/node_modules/faucet/bin/cmd.js

Reply:
since this is "app-spec" it sets a better example for others to look at I think

Reply:
The `mkdir dist` is necessary when it's running on travis since empty directories are not committed to git. This was the last fix for the CI that I figured out after many hours.

Also I'd prefer to have all those stages separately in the CI and have those separators in the output since we will work with the app-spec-output a lot going forward.

We expect most of the PRs to come to fail the CI until we build the features used here. So people will be looking at the CI output a lot. The use case for `hcdev test` is running tests manually which is different. If we need to have a shell script for the CI anyway I don't see why we need to use hcdev there too.